lib/slackify/router.rb

```diff
@@ -23,10 +26,74 @@ def call_command(message, params)
 
           Slackify.configuration.unhandled_handler.unhandled(params)
         else
-          new_params = params.merge(command_arguments: command.regex.match(message).named_captures)
+          new_params = params.merge({ command_arguments: extract_arguments(message, command) })
+
           command.handler.call(new_params)
         end
       end
+
+      def extract_arguments(message, command)
+        if command.regex
+          command.regex.match(message).named_captures
+        else
+          raw_arguments = message.sub(/^#{command.base_command}/, '').strip
+          spec = {}
+          command.parameters.each do |parameter|
+            spec[parameter.keys[0].to_sym] = { type: parameter.values[0].to_sym }
+          end
+          parse_by_spec(spec, raw_arguments)
+        end
+      end
+
+      def parse_by_spec(spec, raw_arguments)
+        processed_args = {}
+
+        s = StringScanner.new(raw_arguments)
+        until s.eos?
+          # get the key, remove '=' and extra whitespace
+          current_key = s.scan_until(/=/)
+          break if current_key.nil?
+
+          current_key = current_key[0..-2].strip
+
+          # grab value accounting for any quotes
+          terminating_string = case s.peek(1)
+                               when "'"
+                                 s.skip(/'/)
+                                 /'/
+                               when '"'
+                                 s.skip(/"/)
+                                 /"/
+                               else
+                                 / /
+                               end
+          processed_args[current_key.to_sym] = if s.exist?(terminating_string)
+                                                 # grab everything before the next instance of the terminating character
+                                                 s.scan_until(terminating_string)[0..-2]
+                                               else
+                                                 # this is probably wrong unless we were expecting a space, but hit eos
+                                                 s.rest
+                                               end
+        end
+
+        # only pass on expected parameters for now.
+        processed_spec = {}
+        spec.each do |key, value|
+          # coerce to the expected type
+          processed_spec[key] = case value.fetch(:type, 'string')
```

Comment:
I feel like this fallback will never get hit. If we don't specify a type and just do

```yaml
parameters:
  - first
  - second
```

we'll break on line 42 as we expect a hash. We could add a rule in the validator to check for a hash with only the 4 types we support at the moment (string, int, float, bool)

Reply:
Agreed it won't be hit, I reworked it in the #17 and I'll add a test now for these core types